### PR TITLE
Fix jsPlumb connection detaching

### PR DESCRIPTION
**Pull request by justinvdm**
At the moment, destination endpoint connection detaches are broken. We've been using our own slightly modified jsPlumb fork with a workaround for this, but the workaround appears to not always work. After a bit of tinkering, it seems that jsPlumb is unhappy if `jsPlumb.detach()` is called on an already detached endpoint.  Fixing that is simple enough -- just infer whether the connection has already been detached before trying to detach it.

### CODE REVIEW

**Comment by justinvdm** [comment]
@SteveBarnett ready for review, a pretty small fix that we've needed for a very long time :)

**Comment by SteveBarnett** [comment]
\o/, :+1: , once Travis is happy.

**Comment by justinvdm** [comment]
@SteveBarnett thanks for the review, it seems this fix doesn't work jsPlumb 1.7.5 (the version I'm planning on upgrading us to), going to keep tinkering.

**Comment by justinvdm** [comment]
Ah, so it seems like a race condition or something -- when setting breakpoints, `.plumbConnection.endpoints` are `null` by the time we check it in `.destroy()`. When breakpoints aren't set, `.plumbConnection.endpoints` aren't `null`. I'm going to go with a solution where we tell `.destroy()` whether the destroy is because of a connection detach done by the user, in which case it should not detach.

Related -- `jsPlumb.detach()` seems to cause endless recursion if we don't set `fireEvent` to `false`, though the docs say `false` is the default. Will make a comment.

**Comment by justinvdm** [comment]
@SteveBarnett actually ready for review now :)

**Comment by hodgestar** [comment]
:+1: from my side.

**Comment by SteveBarnett** [comment]
:+1: from me too.
